crates/zkvm/zkvm/src/proof.rs

```diff
@@ -2,80 +2,98 @@ use arbitrary::Arbitrary;
 use borsh::{BorshDeserialize, BorshSerialize};
 use serde::{Deserialize, Serialize};
 
-/// Validity proof generated by the `ZkVmHost`
+/// Represents a wrapper around a byte vector for various proofs and keys.
+///
+/// Provides common utilities such as byte access and emptiness checks.
 #[derive(
     Debug, Clone, Serialize, Deserialize, BorshSerialize, BorshDeserialize, PartialEq, Eq, Arbitrary,
 )]
-pub struct Proof(Vec<u8>);
+pub struct ByteWrapper(Vec<u8>);
 
-impl Proof {
+impl ByteWrapper {
+    /// Creates a new instance from a `Vec<u8>`.
     pub fn new(data: Vec<u8>) -> Self {
         Self(data)
     }
 
+    /// Returns a reference to the inner byte slice.
     pub fn as_bytes(&self) -> &[u8] {
         &self.0
     }
 
+    /// Checks if the byte vector is empty.
     pub fn is_empty(&self) -> bool {
         self.0.is_empty()
     }
 }
 
-impl From<Proof> for Vec<u8> {
-    fn from(value: Proof) -> Self {
+impl From<ByteWrapper> for Vec<u8> {
+    fn from(value: ByteWrapper) -> Self {
         value.0
     }
 }
 
-impl From<&Proof> for Vec<u8> {
-    fn from(value: &Proof) -> Self {
+impl From<&ByteWrapper> for Vec<u8> {
+    fn from(value: &ByteWrapper) -> Self {
         value.0.clone()
     }
 }
 
-/// Verification Key required to verify proof generated from `ZKVMHost`
+/// Validity proof generated by the `ZkVmHost`.
+pub type Proof = ByteWrapper;
+
+/// Public values associated with a [`Proof`].
+pub type PublicValues = ByteWrapper;
+
+/// Verification Key required to verify proof generated from `ZkVmHost`.
+pub type VerificationKey = ByteWrapper;
+
+/// A receipt containing a `Proof` and associated `PublicValues`.
 #[derive(
     Debug, Clone, Serialize, Deserialize, BorshSerialize, BorshDeserialize, PartialEq, Eq, Arbitrary,
 )]
-pub struct VerificationKey(Vec<u8>);
-
-impl VerificationKey {
-    pub fn new(data: Vec<u8>) -> Self {
-        Self(data)
-    }
-
-    pub fn as_bytes(&self) -> &[u8] {
-        &self.0
-    }
+pub struct ProofReceipt {
+    /// The validity proof.
+    pub proof: Proof,
+    /// The public values associated with the proof.
+    pub public_values: PublicValues,
```

Comment:
I like the idea bu to be honest I think we need to make these fields private and implement a pub getter. What do you think?
I am not that happy with something important as proofs receipt having public fields. I think that it kinda future-proofs it if someone wants to add something else to the `ProofReceipt` since everything is private with public getters.

Reply:
Yup makes sense! Thanks!